File: PyLibs/src/Libs/DataStruct/LockQueue.py
# -*- coding: utf-8 -*-

from Libs.Core.GObject import GObject;
from Libs.DataStruct.LockList import LockList;

class LockQueue(GObject):
    
    def __init__(self, name = ""):
        super(LockQueue, self).__init__();
        
        self.mList = LockList();
        
        self.mName = name;
        

    def push(self, item):
        self.mList.Add(item);


    def pop(self):
        return self.mList.RemoveAt(0);

File: JSLib/src/DataStruct/LockQueue.js
namespace SDK.Lib
{
    public class LockQueue<T>
    {
        protected LockList<T> m_list;

        public LockQueue(string name)
        {
            m_list = new LockList<T>("name");
        }

        public void push(T item)
        {
            m_list.Add(item);
        }

        public T pop()
        {
            return m_list.RemoveAt(0);
        }
    }
}